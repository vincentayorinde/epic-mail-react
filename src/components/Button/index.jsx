import React from 'react';

const Button = ({onClick, classes, type, name, children, isSubmit, ...rest}) => {
  return (
    <button onClick={onClick} type={type} {...rest}>
        { isSubmit || name }
    </button>
  );
};

export default Button;
