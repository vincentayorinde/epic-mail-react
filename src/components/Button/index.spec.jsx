import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

describe('Button Component', () => {
  it('Should render button without errors', () => {
    const component = shallow(<Button />);
    expect(component.find('button').length).toBe(1);
    expect(component).toMatchSnapshot();
  });
  
  it('sets start button to Loading... when isSubmit is true', () =>{
    const component = shallow(<Button name="Start" isSubmit={true} />);
    expect(component.find('button').text()).toBe('Loading...');
  })
})