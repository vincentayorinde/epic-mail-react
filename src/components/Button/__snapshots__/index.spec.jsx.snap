// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render button without errors 1`] = `<button />`;
